filament/backend/src/metal/MetalHandles.mm

```diff
@@ -245,30 +245,53 @@ static inline MTLTextureUsage getMetalTextureUsage(TextureUsage usage) {
     }
 }
 
-struct PresentDrawableData {
-    void* drawable = nullptr;
-    MetalDriver* driver = nullptr;
-};
+#ifndef FILAMENT_RELEASE_PRESENT_DRAWABLE_MAIN_THREAD
+#define FILAMENT_RELEASE_PRESENT_DRAWABLE_MAIN_THREAD 1
+#endif
 
-void presentDrawable(bool presentFrame, void* user) {
-    auto* presentDrawableData = static_cast<PresentDrawableData*>(user);
+class PresentDrawableData {
+public:
+    PresentDrawableData() = delete;
+    PresentDrawableData(const PresentDrawableData&) = delete;
+    PresentDrawableData& operator=(const PresentDrawableData&) = delete;
+
+    static PresentDrawableData* create(id<CAMetalDrawable> drawable, MetalDriver* driver) {
+        assert_invariant(driver);
+        return new PresentDrawableData(drawable, driver);
+    }
+
+    void maybePresentAndCleanupAsync(bool shouldPresent) {
```

Comment:
I would prefer this method to be static and take a PresentDrawableData* (or &), because when it returns the object is effectively destroyed. That would be clearer. Also I would rename to `maybePresentAndDestroyAsync`.